views/expenses/index.ejs

```diff
@@ -108,8 +108,8 @@
             </thead>
             <tbody class="">
                 <% for (expense of expenses) { %> <%
-                if(expense.category.parentCategory?.name === category.name ||
-                expense.category.name === category.name) { %>
+                if(expense.category.parentCategory.id === category.id ||
```

Comment:
OK, here's what's happening:
- we are going through all expenses and listing the ones that belong in the current category
- we have to check both `expense.category.parentCategory` and `expense.category`. This is because we added support for putting expenses directly in the main category at some point. For example:
   - If an expense is in `BMW - registracija`, then `expense.category.parentCategory === category)` will be `true`
   - If an expense is directly in `BMW`, then `expense.category === category` will be `true`
- The problem is that we were comparing category _names_, which can be duplicate (e.g. you have a main category _Ostalo_ and a subcategory _Ostalo_ under _Hrana_)
- The solution is to compare _ids_, which are guarenteed to be unique

Reply:
OK, I get it now. I guess it's almost always better to work with IDs.